cmd/tier: panic/error on pull

```
jevon@Jevons-MacBook-Air tierx % go run tier.run/cmd/tier@oauth pull
panic: runtime error: slice bounds out of range [22:21]

goroutine 1 [running]:
golang.org/x/exp/slices.Delete[...]({0x1400033e200?, 0x14000120000?, 0x1?}, 0x5000?, 0x1?)
	/Users/jevon/go/pkg/mod/golang.org/x/exp@v0.0.0-20220722155223-a9213eeb770e/slices/slices.go:157 +0xf8
main.filterNonTierPlans(...)
	/Users/jevon/go/pkg/mod/tier.run@v0.1.1-0.20220907000054-239344c14a7c/cmd/tier/tier.go:208
main.tier({0x16d537af2?, 0x14000132010?}, {0x14000132010?, 0x14000002340?, 0x14000044768?})
	/Users/jevon/go/pkg/mod/tier.run@v0.1.1-0.20220907000054-239344c14a7c/cmd/tier/tier.go:113 +0x4f0
main.main()
	/Users/jevon/go/pkg/mod/tier.run@v0.1.1-0.20220907000054-239344c14a7c/cmd/tier/tier.go:39 +0x120
exit status 2
```

--- a/cmd/tier/tier.go
+++ b/cmd/tier/tier.go
@@ -12,7 +12,6 @@ import (
 	"os"
 	"text/tabwriter"
 
-	"golang.org/x/exp/slices"
 	"tier.run/cmd/tier/profile"
 	"tier.run/pricing"
 	"tier.run/pricing/schema"
@@ -199,15 +198,11 @@ func tc() *pricing.Client {
 }
 
 func filterNonTierPlans(plans schema.Plans) schema.Plans {
-	for i, p := range plans {
-		if p.ID == "" {
-			plans = slices.Delete(plans, i, i+1)
-		}
-		for j, f := range p.Features {
-			if f.Err != nil {
-				p.Features = slices.Delete(p.Features, j, j+1)
-			}
+	var dst schema.Plans
+	for _, p := range plans {
+		if p.ID != "" {
+			plans = append(dst, p)
 		}
 	}
-	return plans
+	return dst
 }

--- a/cmd/tier/tier.go
+++ b/cmd/tier/tier.go
@@ -118,7 +118,7 @@ func tier(cmd string, args []string) error {
 			return err
 		}
 
-		filterNonTierPlans(m.Plans)
+		m.Plans = filterNonTierPlans(m.Plans)
 
 		out, err := json.MarshalIndent(m, "", "     ")
 		if err != nil {
@@ -133,7 +133,7 @@ func tier(cmd string, args []string) error {
 			return err
 		}
 
-		filterNonTierPlans(m.Plans)
+		m.Plans = filterNonTierPlans(m.Plans)
 
 		tw := tabwriter.NewWriter(stdout, 0, 2, 2, ' ', 0)
 		defer tw.Flush()
@@ -201,7 +201,7 @@ func filterNonTierPlans(plans schema.Plans) schema.Plans {
 	var dst schema.Plans
 	for _, p := range plans {
 		if p.ID != "" {
-			plans = append(dst, p)
+			dst = append(dst, p)
 		}
 	}
 	return dst

